src/components/infinite-scroll/infinite-scroll.ts

```diff
@@ -267,7 +286,23 @@ export class InfiniteScroll {
    * to `enabled`.
    */
   complete() {
-    if (this.state === STATE_LOADING) {
+    if (this._position === POSITION_TOP) {
```

Comment:
Sure! I'll add a few more comments as well.

New content is being added at the top, but the `scrollTop` position stays the same, which causes a scroll jump visually. This algorithm makes sure to prevent this.

1. `complete()` is called, but the UI hasn't had time to update yet.
2. Save the current content dimensions.
3. Wait for the next frame using `_dom.read`, so the UI will be updated.
4. Read the new content dimensions.
5. Calculate the height difference and the new scroll position.
6. Delay the scroll position change using `_dom.write` to be performant.
7. Change the scroll position (= visually maintain the scroll position).
8. Change the state to re-enable the InfiniteScroll. This should be after changing the scroll position, or it could cause the InfiniteScroll to be triggered again immediately.